src/main/java/com/google/sps/data/GivrUser.java

```diff
@@ -51,53 +56,97 @@ public boolean isMaintainer() {
     return this.isMaintainer;
   }
 
-  //TODO(): get correct moderator status for organization
+  public String getUserEmail() {
+    return this.email;
+  }
+
+  public boolean isLoggedIn() {
+    return this.isLoggedIn;
+  }
+
+  // TODO(): get correct moderator status for organization
   public boolean isModerator(long organizationId) {
       return false;
   }
 
-  public static GivrUser getUserByIdFromDatastore(String userId) {
+  // Gets User with propertyName, propertyValue exists within Datastore.
+  public static Entity getUserFromDatastoreWithProperty(String propertyName, String propertyValue) {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Filter queryFilter = new FilterPredicate("userId", FilterOperator.EQUAL, userId);
+
+    Filter queryFilter = new FilterPredicate(propertyName, FilterOperator.EQUAL, propertyValue);
     Query query = new Query("User").setFilter(queryFilter);
-      
-    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(1);
     PreparedQuery preparedQuery = datastore.prepare(query);
-    QueryResultList<Entity> userResult = preparedQuery.asQueryResultList(fetchOptions);
+    Entity entity = preparedQuery.asSingleEntity();
```

Comment:
IIRC this throws if there is more than one result. You can decide if you want to handle that here or not

https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery